### PR TITLE
Added Instructor In Tickets And Enables admin to edit the ticket

### PR DESCRIPTION
**Pull request by jehanxaibahmed**
Please review it and merge .

### CODE REVIEW

File: app/Http/Controllers/TicketsController.php [hunk 1]
```diff
@@ -34,8 +39,17 @@ public function create()
     {
         // Gate::authorize('create_ticket');
         $categories = TicketCategory::all();
+        $instructors = User::select('users.*', 'roles.name', 'users.name as user_name')
+        ->join('model_has_roles', 'users.id', '=', 'model_has_roles.model_id')
+        ->join('roles', 'model_has_roles.role_id', '=', 'roles.id')
+        ->where('roles.name', '=', 'Teacher')->get();
```
Comment (guardianstraining) [inline app/Http/Controllers/TicketsController.php:45]: There's a helper function for this in the user model.

User::Teacher()->get() should return all users with the role of a teacher thus you do not need this block of code

File: app/Http/Controllers/TicketsController.php [hunk 2]
```diff
@@ -105,11 +121,38 @@ public function close($ticket_id, AppMailer $mailer)
...
         return redirect()->back()->with("status", "The ticket has been closed.");
     }
 
+
+    public function edit($ticket_id)
+    {
+        $ticket = Ticket::where('ticket_id', $ticket_id)->firstOrFail();
+        $categories = TicketCategory::all();
+        $instructors = User::select('users.*', 'roles.name', 'users.name as user_name')
+        ->join('model_has_roles', 'users.id', '=', 'model_has_roles.model_id')
```
Comment (guardianstraining) [inline app/Http/Controllers/TicketsController.php:137]: Previous comment applies here

File: app/Http/Requests/UpdateTicketRequest.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        return [
+            // 'instructor_id' => 'required',
+            // 'category_id' => 'required',
+            // 'ticket_id' => 'required',
```
Comment (guardianstraining) [inline app/Http/Requests/UpdateTicketRequest.php:29]: Validation should check for foreign keys

File: app/Http/Requests/UpdateTicketRequest.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            // 'title' => 'required',
+            // 'priority' => 'required',
+            'status' => 'required',
```
Comment (guardianstraining) [inline app/Http/Requests/UpdateTicketRequest.php:32]: There's an enum for status. Ignore if you do not have that update yet

File: resources/views/jambasangsang/backend/tickets/update.blade.php [hunk 5]
```diff
@@ -0,0 +1,218 @@
+@extends('layouts.backend.master')
+@section('content')
+
+@section('breadcrumbs')
+    <div class="breadcrumbs">
+        <div class="col-sm-4">
+            <div class="page-header float-left">
+                <div class="page-title">
+                    <h1>Edit Support Ticket</h1>
```
Comment (guardianstraining) [inline resources/views/jambasangsang/backend/tickets/update.blade.php:9]: There's a livewire breadcrumb component, you can make use of that

File: resources/views/jambasangsang/backend/tickets/update.blade.php [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+
+                    <div class="col-md-6">
+                        <div class="form-group{{ $errors->has('category') ? ' has-error' : '' }}">
+                            <p>Category</p>
+
+                            <select id="category" type="category" class="form-control" name="category">
+                                <option value="">Select Category</option>
+                                @foreach ($categories as $category)
+                                    <option value="{{ $category->id }}"
```
Comment (guardianstraining) [inline resources/views/jambasangsang/backend/tickets/update.blade.php:80]: There's a livewire category select, please use that

File: resources/views/jambasangsang/backend/tickets/update.blade.php [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+
+                    <div class="col-md-6">
+                        <div class="form-group{{ $errors->has('instructor') ? ' has-error' : '' }}">
+                            <p>Instructor</p>
+
+                            <select id="instructor" type="instructor" class="form-control" name="instructor">
+                                <option value="">Select Instructor</option>
+                                @foreach ($instructors as $instructor)
+                                    <option value="{{ $instructor->id }}"
```
Comment (guardianstraining) [inline resources/views/jambasangsang/backend/tickets/update.blade.php:104]: There's a livewire teacher select, please use that

File: resources/views/jambasangsang/backend/user_tickets/create.blade.php [hunk 8]
```diff
@@ -89,6 +89,32 @@
 
 
 
+                        <div class="col-md-6">
+                            <div class="form-group{{ $errors->has('instructor') ? ' has-error' : '' }}">
+                                <p>Instructor</p>
+
+                                    <select id="instructor" type="instructor" class="form-control" name="instructor">
+                                        <option value="">Select Instructor</option>
+                                        @foreach ($instructors as $instructor)
```
Comment (guardianstraining) [inline resources/views/jambasangsang/backend/user_tickets/create.blade.php:98]: See livewire component for this

**Review by guardianstraining — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/TicketsController.php (hunk 1 above)
Comment (jehanxaibahmed) [reply app/Http/Controllers/TicketsController.php]: Resolved.

File: resources/views/jambasangsang/backend/tickets/update.blade.php (hunk 6 above)
Comment (jehanxaibahmed) [reply resources/views/jambasangsang/backend/tickets/update.blade.php]: Resolved

File: resources/views/jambasangsang/backend/tickets/update.blade.php (hunk 7 above)
Comment (jehanxaibahmed) [reply resources/views/jambasangsang/backend/tickets/update.blade.php]: Resolved

File: resources/views/jambasangsang/backend/user_tickets/create.blade.php (hunk 8 above)
Comment (jehanxaibahmed) [reply resources/views/jambasangsang/backend/user_tickets/create.blade.php]: Resolved

**Review by guardianstraining — CHANGES_REQUESTED** [review]
- When I create ticket it redirects back to same form without any response
- Update ticket does not account for existing unique name as I cannot update a ticket with same name
- Delete ticket does not return any response
- Use the changelog-md or add a good commit description that contains a list of what you did so it's easy to review

**Comment by guardianstraining** [comment]
There's no need to change the teacher_id to instructor_id as you'd either have to work with it or make the change across the entire platform

**Pull request merged**